php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -896,4 +895,24 @@ public function byteSize()
         }
         return $size;
     }
+
+    private function appendHelper($field, $append_value)
+    {
+        $getter = $field->getGetter();
+        $setter = $field->getSetter();
+
+        $field_arr_value = $this->$getter();
+        $field_arr_value[] = $append_value;
+        $this->$setter($field_arr_value);
+    }
+
+    private function kvUpdateHelper($field, $update_key, $update_value)
+    {
+        $getter = $field->getGetter();
+        $setter = $field->getSetter();
+
+        $field_arr_value = $this->$getter();
+        $field_arr_value[$update_key] = $update_value;
+        $this->$setter($field_arr_value);
```

Comment:
However, the $field_arr_value returned on line914 is a MapField object instead of array.